force-app/main/default/lwc/barcodeScannerExample/__tests__/barcodeScannerExample.test.js

```diff
@@ -0,0 +1,183 @@
+import { createElement } from 'lwc';
+import { getNavigateCalledWith } from 'lightning/navigation';
+import BarcodeScannerExample from 'c/barcodeScannerExample';
+
+// Mock various barcode functionality from mobileCapabilites.js
+import {
+    resetBarcodeScannerStubs,
+    setBarcodeScannerAvailable,
+    setUserCanceledScan,
+    setBarcodeScanError
+} from 'lightning/mobileCapabilities';
+
+// Mock various barcode functionality from platformShowToastEvent.js
+import {
+    resetAllShowToastEventStubs,
+    showToastEventCalledWith
+} from 'lightning/platformShowToastEvent';
+
+describe('c-barcode-scanner-example', () => {
+    afterEach(() => {
+        // Reset the JSDOM instance shared across test cases in a single file
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+
+        // Prevent data saved on mocks from leaking between tests
+        jest.clearAllMocks();
+
+        // Reset stubs
```

Comment:
Am I correct in understanding the following helper values would not be reset by `clearAllMocks`?

Reply:
Yes. `jest.clearAllMocks()` will only affect functions declared with `jest.fn()`.